OPSEXP-2486 Rewrite ingress, dns and https docs for eks deployment (#1138)
Alfresco/acs-deployment: docs/helm/eks-deployment.md

EKS_CLUSTER_NAME=my-alfresco-eks
```

Create the cluster using the latest supported version, check the main [README](../../README.md).
Most common choices for instance types are `m5.xlarge` and `t3.xlarge`:

```sh

> eksctl create cluster --name $EKS_CLUSTER_NAME --version 1.29 --instance-types t3.xlarge --nodes 3

Reviewer: Shouldn't it be 1.28 as we state we test only with EKS 1.28?

Author: strictly speaking yes, but given that I tested it manually during this PR and we are testing 1.29 on KinD... we could stay with that

> * Use an Elastic File System, installing the ([EFS CSI driver](#efs-csi-driver))
>   (the default we suggest, see `storageClass="nfs-client"` values in [helm
>   install section](#latest-enterprise-version) and required when repository
>   replicas > 1)

Reviewer: ```suggestion
* Use an Elastic File System, installing the [EFS CSI driver](#efs-csi-driver) (this is what
  is documented below)
```

> * Use an EBS block-storage, enabling [EBS CSI driver](#ebs-csi-driver) (when
>   repository replicas = 1)

Reviewer: ```suggestion
* Use an EBS block-storage, enabling [EBS CSI driver](#ebs-csi-driver) (only
  valid if there's a single replica and if pods cannot be scheduled in different
  availability zones. Avoid in production)
```

```sh
eksctl create iamserviceaccount \
--name ebs-csi-controller-sa \
--namespace kube-system \
--cluster $EKS_CLUSTER_NAME \
--attach-policy-arn arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy \
--approve \
--role-only \
--role-name AmazonEKS_EBS_CSI_DriverRole
```

Enable the addon referencing the IAM role created previously:

```sh
eksctl create addon \
--name aws-ebs-csi-driver \
--cluster $EKS_CLUSTER_NAME \
--service-account-role-arn arn:aws:iam::${AWS_ACCOUNT_ID}:role/AmazonEKS_EBS_CSI_DriverRole \
--force
```

At this point the provisioning of EBS volumes using the default GP2
storageClass will be handled by this driver.

For further information please refer to the official
[Amazon EBS CSI driver](https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html) guide.

## Deploy

Now the EKS cluster is setup we can start to provision Kubernetes resources on
top of it.

### Ingress

See [ingress-nginx](ingress-nginx.md) section.

### DNS

In order to access Alfresco once installed, you need to setup a DNS record that
resolve to the ELB hostname that has been provisioned by `ingress-nginx`.

> This is a typical architecture for which you can learn more at

Reviewer 2: ```suggestion
This is typical architecture, which you can learn more about at
```

[Exposing Kubernetes Applications article](https://aws.amazon.com/blogs/containers/exposing-kubernetes-applications-part-3-nginx-ingress-controller/).

To retrieve the automatically assigned hostname of the ELB you need to inspect
the ingress resources:

```sh
kubectl get service -n ingress-nginx
```

which will return an output like the following:

```sh
NAME                       TYPE           CLUSTER-IP       EXTERNAL-IP                       PORT(S)                      AGE
ingress-nginx-controller   LoadBalancer   10.100.132.184   ???.eu-west-1.elb.amazonaws.com   80:31780/TCP,443:32152/TCP   3m
```

Now you can proceed creating a new DNS record in `YOUR-DOMAIN-NAME` zone, like:

* Record name: `acs`
* Record type: `CNAME`
* Value: `???.eu-west-1.elb.amazonaws.com`

Wait a few minutes before trying to access `http://acs.YOUR-DOMAIN-NAME` in your
browser to let the new record to propagate. Once ready, you should get the
default `404 Not Found` nginx error page.

Set an environment variable with the hostname which will be useful later.

```sh
export ACS_HOSTNAME=acs.YOUR-DOMAIN-NAME
```

### HTTPS

> The most simple way to access applications running on Kubernetes behind HTTPS is

Reviewer: ```suggestion
The simplest way to access applications running on Kubernetes behind HTTPS is
```

`YOUR-DOMAIN-NAME` with the hosted zone you created earlier and `MAJOR` &
`MINOR` with the appropriate values):

```sh
helm upgrade --install acs alfresco/alfresco-content-services \
--values=MAJOR.MINOR.N_values.yaml \
--set alfresco-repository.persistence.enabled=true \
--set alfresco-repository.persistence.storageClass="nfs-client" \
--set alfresco-transform-service.filestore.persistence.enabled=true \
--set alfresco-transform-service.filestore.persistence.storageClass="nfs-client" \
--set global.known_urls=https://${ACS_HOSTNAME} \
--set global.alfrescoRegistryPullSecrets=quay-registry-secret \
--values letsencrypt_values.yaml \
--namespace=alfresco
```

### Wait for successfull deployment

You can monitor the progress of deployments with:

```sh
kubectl get pod -n alfresco
```

In a few minutes, each pod should be in `Running` in the `Status` column and
showing `1/1` in the `Ready` column.

If it doesn't happen, you can first describe the pod not in Running state yet
and look for the events section at the end:

```sh
kubectl describe pod acs-alfresco-repository-???-??? -n alfresco
```

If the pod is in running state but can't achieve the `1/1` Ready before the

> liveness timeout kicks in, you should take a look at the logs with:

Reviewer: ```suggestion
readiness probe fails as many times the readiness threshold allows, you should
take a look at the logs with:
```
